packages/app-admin-ui/client/pages/Item/index.js

```diff
@@ -55,6 +55,30 @@ const getCurrentValues = memoizeOne(getValues);
 
 const deserializeItem = memoizeOne((list, data) => list.deserializeItemData(data));
 
+const getFieldGroups = memoizeOne(list => {
+  const fieldGroups = list.fieldGroups || [];
+  if(fieldGroups.length) {
```

Comment:
logic for `getRenderableFields()` is missing in this condition. fields should be filtered in the group based on output of `getRenderableFields`, also the empty groups should be filtered out.

Reply:
I think my conf format is more comfortable for developers :) Because you quickly see how many groups are there + we can add new fields to the config like:
```
    {
      label: () => i18.t('Misc') // label resolver function
      fields: ['maxRsvps', 'isRsvpAvailable', 'themeColor'],
      isCollapsible: true, // can be collapsed or not
      collapsed: true // default collapsed state
    },
```

Reply:
Thanks @kschingiz . I like this.

Please consider the following suggestions
``` javascript
fieldGroups: [
  // fieldGroups to be displayed in the same sequence as they are mentioned in the array
  {
    label: 'Base',
    fieldsInSequence: ['name', 'status', 'description', 'startTime']
  },
  {
    label: 'Location',
    fieldsInSequence: ['locationAddress', 'locationDescription']
  },
  {
    label: () => i18.t('Misc') // label resolver function
    fieldsInSequence: ['maxRsvps', 'isRsvpAvailable', 'themeColor'], 
    // Changed to indicate that the fields will be shown in the UI in the sequence specified
    isCollapsible: true, // can be collapsed or not
    collapsed: () => { return true } , // function that returns true or false
    displayFieldGroup: () => {  
      //  function that returns true or false
      //  This logic can be based on fields filled in some other group
      //  e.g. If a person is single, do not ask details of the spouse, 
      //  the spouseFieldGroup is not displayed
      if ( /*  logic to check if field group has to be displayed */ ) {
        return true;
      } else {
	return false
      }, 
    },
]
```

Reply:
Another thought: Should the fieldGroups be part of the UI block as these are related to the UI / display?

Reply:
```
UI customization would be handled in separate file ./admin-ui/index.js
```
@gautamsi  I am new to keystone, do we have any UI customization already, if yes I can refactor it so we don't clutter schema. How to use this admin-ui/index.js file? I cannot find any references to it.